xbmc/music/Artist.h

```diff
@@ -130,21 +132,49 @@ class CArtistCredit
 
   std::string GetArtist() const                { return m_strArtist; }
   std::string GetMusicBrainzArtistID() const   { return m_strMusicBrainzArtistID; }
-  std::string GetJoinPhrase() const            { return m_strJoinPhrase; }
   int         GetArtistId() const              { return idArtist; }
   void SetArtist(const std::string &strArtist) { m_strArtist = strArtist; }
   void SetMusicBrainzArtistID(const std::string &strMusicBrainzArtistID) { m_strMusicBrainzArtistID = strMusicBrainzArtistID; }
-  void SetJoinPhrase(const std::string &strJoinPhrase) { m_strJoinPhrase = strJoinPhrase; }
   void SetArtistId(int idArtist)               { this->idArtist = idArtist; }
 
 private:
   long idArtist;
   std::string m_strArtist;
   std::string m_strMusicBrainzArtistID;
-  std::string m_strJoinPhrase;
-  bool m_boolFeatured;
 };
 
 typedef std::vector<CArtist> VECARTISTS;
 typedef std::vector<CArtistCredit> VECARTISTCREDITS;
 
+#define ROLE_ARTIST 1  //Default role
+
+class CMusicRole
+{
+public:
+  CMusicRole() { }
+  CMusicRole(std::string strRole, std::string strArtist) : idRole(-1), m_strRole(strRole), m_strArtist(strArtist), idArtist(-1) { }
+  CMusicRole(int role, std::string strRole, std::string strArtist, long ArtistId) : idRole(role), m_strRole(strRole), m_strArtist(strArtist), idArtist(ArtistId) { }
```

Comment:
Going to stay with long (for now) to match how artistId is treated in the rest of artist.h, however I think we need to review the type for artistId and albumId. It is long in some places, but most of musicdatabase.cpp processes them as int. Probably needs to be int64_t everywhere.